frontend/src/screens/apps/NewApp.tsx

```diff
@@ -260,13 +261,41 @@ const NewAppInternal = ({ capabilities }: NewAppInternalProps) => {
 
       const createAppResponse = await createApp(createAppRequest);
 
+      // dispatch a success event which can be listened to by the opener or by the app that embedded the webview
+      // this gives those apps the chance to know the user has enabled the connection
+      const nwcEvent = new CustomEvent("nwc:success", {
+        detail: {
+          relayUrl: createAppResponse.relayUrl,
+          walletPubkey: createAppResponse.walletPubkey,
+          lud16: createAppResponse.lud16,
+        },
+      });
+      window.dispatchEvent(nwcEvent);
+
+      // notify the opener of the successful connection
+      if (window.opener) {
+        window.opener.postMessage(
+          {
+            type: "nwc:success",
+            relayUrl: createAppResponse.relayUrl,
+            walletPubkey: createAppResponse.walletPubkey,
+            lud16: createAppResponse.lud16,
+          },
+          "*"
+        );
+      }
+
       if (createAppResponse.returnTo) {
         // open connection URI directly in an app
         window.location.href = createAppResponse.returnTo;
         return;
       }
       toast("App created");
       setCreateAppResponse(createAppResponse);
+      if (pubkey) {
```

Comment:
I am not sure we should add this navigation. For most cases either there will be a return_to or a listener for one of the success messages above (like Bitcoin Connect, which will then close the dialog)